roles/kubernetes-apps/ansible/templates/nodelocaldns-daemonset.yml.j2

```diff
@@ -47,17 +47,18 @@ spec:
         - coredns
 {% if enable_nodelocaldns_secondary %}
         - -skipteardown
-{% else %}
+{% endif %}
         ports:
+        - containerPort: {{ nodelocaldns_prometheus_port }}
+          name: metrics
+          protocol: TCP
+{% if not enable_nodelocaldns_secondary %}
         - containerPort: 53
           name: dns
           protocol: UDP
         - containerPort: 53
           name: dns-tcp
           protocol: TCP
-        - containerPort: 9253
-          name: metrics
-          protocol: TCP
```

Comment:
Sorry to not have commented on that before:
Is there a reason to have this conditional at all ? I mean, I don't get why we should not define the ports when using coredns-dual, and neither 039205560a5a38dac7e180ec4c46ce4edd404d39 nor #8100 really answer that.

@cristicalin maybe you remember something about this ?

Reply:
How do you want me to proceed?

Reply:
I think you can just drop the if-else and unconditionally define the ports.

Reply:
Applied. PTAL.